src/dataflow/src/source/mod.rs

```diff
@@ -436,365 +427,6 @@ impl fmt::Debug for SourceMessage<Option<Vec<u8>>, Option<Vec<u8>>> {
-            if !self.knows_of(&pid) && self.responsible_for(&pid) {
-                source.add_partition(pid.clone(), offset.clone());
-                self.add_partition(&pid, offset);
-            }
-
-            if !self.responsible_for(&pid) {
-                continue;
-            }
-            let cons_info = self
-                .partition_metadata
-                .get_mut(&pid)
-                .expect("partition known to exist");
-
-            if let Some((timestamp, max)) =
-                timestamp_bindings.get_binding(&pid, cons_info.offset() + 1)
-            {
-                cons_info.update_timestamp(timestamp, max);
-            }
-        }
-    }
-
-    fn downgrade_capability(
-        &mut self,
-        cap: &mut Capability<Timestamp>,
-        timestamp_bindings: &mut TimestampBindingRc,
-    ) {
-        //  We need to determine the maximum timestamp that is fully closed. This corresponds to the minimum of
-        //  * closed timestamps across all partitions we own
-        //  * maximum bound timestamps across all partitions we don't own (the `upper`)
-        let mut upper = Antichain::new();
-        timestamp_bindings.read_upper(&mut upper);
-
-        let mut min: Option<Timestamp> = if let Some(time) = upper.elements().get(0) {
-            Some(time.saturating_sub(1))
-        } else {
-            None
-        };
-        // Determine which timestamps have been closed. A timestamp is closed once we have processed
-        // all messages that we are going to process for this timestamp across all partitions that the
-        // worker knows about (i.e. the ones the worker has been assigned to read from).
-        // In practice, the following happens:
-        // Per partition, we iterate over the data structure to remove (ts,offset) mappings for which
-        // we have seen all records <= offset. We keep track of the last "closed" timestamp in that partition
-        // in next_partition_ts
-
-        for (pid, cons_info) in self.partition_metadata.iter_mut() {
-            let closed_ts = cons_info.get_closed_timestamp();
-            let metrics = self
-                .source_metrics
-                .partition_metrics
-                .get_mut(&pid)
-                .expect("partition known to exist");
-
-            metrics.closed_ts.set(closed_ts);
-
-            min = match min.as_mut() {
-                Some(min) => Some(std::cmp::min(closed_ts, *min)),
-                None => Some(closed_ts),
-            };
-        }
-        let min = min.unwrap_or(0);
-
-        // Downgrade capability to new minimum open timestamp (which corresponds to min + 1).
-        if min > self.last_closed_ts {
-            self.source_metrics.capability.set(min);
-            cap.downgrade(&(&min + 1));
-            self.last_closed_ts = min;
-
-            let new_compaction_frontier = Antichain::from_elem(min + 1);
-            timestamp_bindings.set_compaction_frontier(new_compaction_frontier.borrow());
```

Comment:
I think this is where the deletion happened. I don't see a corresponding `set_compaction_frontier` being added, but my UI tools may be failing me.